layersvt/device_simulation.cpp

```diff
@@ -1822,6 +1754,89 @@ VKAPI_ATTR VkResult VKAPI_CALL GetPhysicalDeviceToolPropertiesEXT(VkPhysicalDevi
     return result;
 }
 
+VKAPI_ATTR VkResult VKAPI_CALL EnumeratePhysicalDevices(VkInstance instance, uint32_t *pPhysicalDeviceCount, VkPhysicalDevice* pPhysicalDevices) {
+    // Our layer-specific initialization...
+
+    std::lock_guard<std::mutex> lock(global_lock);
+    const auto dt = instance_dispatch_table(instance);
+    VkResult result = dt->EnumeratePhysicalDevices(instance, pPhysicalDeviceCount, pPhysicalDevices);
+
+    // HACK!! epd_count is used to ensure the following code only gets called _after_ vkCreateInstance finishes *in the "vkcube + devsim" use case*
+    if (epd_count > 1 && !pdd_initialized && (VK_SUCCESS == result)) {
```

Comment:
`epd_count` checks to see when `EnumeratePhysicalDevices` calls from the validation layer ([here](https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/f0634ebbb36ffe1cee4372be73bfe8ea8b6ec122/layers/parameter_validation_utils.cpp#L183) and [here](https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/f0634ebbb36ffe1cee4372be73bfe8ea8b6ec122/layers/parameter_validation_utils.cpp#L186)) have finished.

I verified on my Windows AMD machine that this code _without_  the `epd_count` check and _without_ validation layers enabled works.